prqlc/prql-compiler/src/semantic/resolver/transforms.rs

```diff
@@ -238,6 +238,14 @@ impl Resolver<'_> {
 
                 let [pattern, value] = unpack::<2>(func.args);
 
+                if let ExprKind::Array(mut in_values) = pattern.kind {
+                    in_values.insert(0, value);
+                    return Ok(Expr::new(ExprKind::RqOperator {
+                        name: "std.in".to_string(),
+                        args: in_values.clone(),
+                    }));
```

Comment:
> This is fine :D If you want more idiomatic Rust, some of these might look better

Yes totally! I first wanted your input on the prepended column expression in args. Glad it's ok as a first step

> This is a bit of work, so I'm fine with merging this PR with a comment saying "TODO: instead of unpacking the array, compile it to rq::ExprKind::Array".

Thank you for the very clear explanation. Much clearer to me.
For now I added the TODO comment but I'm more than happy to try to work on it this week